plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -965,6 +1014,25 @@ public void executeTableExecute(ConnectorSession session, ConnectorTableExecuteH
         }
     }
 
+    private void executeDropExtendedStats(ConnectorSession session, IcebergTableExecuteHandle executeHandle)
+    {
+        checkArgument(executeHandle.getProcedureHandle() instanceof IcebergDropExtendedStatsHandle);
+
+        Table icebergTable = catalog.loadTable(session, executeHandle.getSchemaTableName());
+
+        verify(transaction == null, "transaction already set");
+        transaction = icebergTable.newTransaction();
```

Comment:
nit: you may extract
```
    private void startTransaction(Supplier<Transaction> transactionSupplier)
    {
        verify(transaction == null, "transaction already set");
        transaction = transactionSupplier.get();
    }
```
and use it here and in other places.

Reply:
pre-existing, let me choose to ignore here, but cc @alexjo2144 @findinpath for potential follow-up